lib/tasks/inventory.rake

```diff
@@ -20,6 +20,32 @@ end
 
 namespace :atc do
   namespace :inventory do
+
+    desc 'Updates the path of SourceObjects via old_path and new_path values in the given path_update_csv_file'
+    task update_source_object_path: :environment do
+      unless ENV['path_update_csv_file']
+        puts 'Please supply a path_update_csv_file'
+        next
+      end
+
+      CSV.foreach(ENV['path_update_csv_file'], headers: true).each do |row|
+        old_path = row['old_path']
+        new_path = row['new_path']
+
+        source_object = SourceObject.for_path(old_path)
+        print "Updating SourceObject #{source_object.id}..."
+        raise "Could not find readable file at: #{new_path}" unless File.readable?(new_path)
+
+        source_object.path = new_path
+        source_object.assign_path_hash
+        source_object.save(validate: false) # Validations normally prevent reassignment of path
+
+        # Verify that object is findable via new_path
+        raise "Cannot find updated SourceObject with new path" unless SourceObject.for_path(new_path)&.id == source_object.id
+        puts "done!"
+      end
+    end
```

Comment:
These are edge cases that should not happen and may not have to be handled, but I want to bring them up:
-) no source_object found for old_path
-) a source_object with new_path actually already exists in ATC before the attempted update.

Reply:
Good news about both cases:

- For `no source_object found for old_path`, that's actually a likely case because we will sometimes move files to their new locations before running this script, so it's okay not to enforce old_path existence.

- For `source_object with new_path actually already exists in ATC`, we have a unique index on the path_hash database column, so that will protect us from accidentally pointing two sources to the same new_path (even when validations are skipped here).

Reply:
Just pushed a commit to better handle case 1.